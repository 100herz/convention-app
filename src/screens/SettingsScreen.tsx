import React from 'react'
import { ScrollView, StyleSheet, View } from 'react-native'

import SettingsList from '@components/Lists/SettingsList'
import LegalList from '@components/Lists/LegalList'
import { defaultStyles, DefaultStyles } from '@styles/theme'
import { Legal } from '@models/legal'
import { Setting } from '@models/setting'

export const settingArray: Setting[] = [{ id: 1, name: 'Push Mitteilungen' }]

const legalScreens: Legal[] = [
  { id: 1, name: 'Impressum' },
  { id: 2, name: 'Datenschutz' },
]

const SettingsScreen: React.FC = () => {
  return (
    <View style={styles.container}>
      <ScrollView>
        <SettingsList data={settingArray} />
        <LegalList data={legalScreens} />
      </ScrollView>
    </View>
  )
}

type Styles = DefaultStyles

const styles = StyleSheet.create<Styles>({
  ...defaultStyles,
  container: {
    ...defaultStyles.container,
    paddingHorizontal: 25,
  },
})

export default SettingsScreen
