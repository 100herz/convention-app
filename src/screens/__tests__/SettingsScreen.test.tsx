import React from 'react'
import { render } from '@testing-library/react-native'

import SettingsScreen from '../SettingsScreen'

describe('<SettingsScreen />', () => {
  it('renders correctly', () => {
    // const { toJSON } = render(<SettingsScreen />)
    // expect(toJSON()).toMatchSnapshot()
  })
})
