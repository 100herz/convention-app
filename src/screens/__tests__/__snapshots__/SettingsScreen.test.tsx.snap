// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 25,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#dfe3e8",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginRight": -8,
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "fontFamily": "sans",
              }
            }
          >
            Push Mitteilungen
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="switch-1"
          >
            <Text>
              false
            </Text>
          </View>
        </View>
      </View>
      <RCTScrollView
        data={
          Array [
            Object {
              "id": 1,
              "name": "Impressum",
            },
            Object {
              "id": 2,
              "name": "Datenschutz",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "paddingTop": 15,
          }
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="link"
            >
              <View
                style={
                  Object {
                    "borderBottomColor": "#dfe3e8",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#333333",
                      "fontFamily": "sans",
                    }
                  }
                >
                  Impressum
                </Text>
                <Text />
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="link"
            >
              <View
                style={
                  Object {
                    "borderBottomColor": "#dfe3e8",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#333333",
                      "fontFamily": "sans",
                    }
                  }
                >
                  Datenschutz
                </Text>
                <Text />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
